src/main/java/com/spikes2212/path/Path.java

```diff
@@ -0,0 +1,25 @@
+package com.spikes2212.path;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class Path {
+    private List<Waypoint> points;
+    public Path(Waypoint... points) {
+        this.points = Arrays.asList(points);
+    }
+
+    public void fill(int middlePoints) {
+        for (int i = 0; i < points.size(); i++) {
```

Comment:
This loop is runs to infinity and causes `OutOfMemoryException`. This happens because the size of points is always larger than the `i`. one way to fix this would be for this method to be static, receive an initial `List<Waypoint>` and then return a new list with filled points, instead of it mutating the existing points in the path.